sjb/generate.py

```diff
@@ -99,11 +99,29 @@ def load_configuration(config_path):
 
     # next, repositories will be synced to the remote VM
     sync_actions = []
+    repo_names=[repository["name"] for repository in job_config.get("sync_repos", [])]
+
+    # determine if test is a pull request
+    is_pull_request = False
+    for repository in job_config.get("sync_repos", []):
+        if repository.get("type"):
+            is_pull_request = True
+
+    def get_dependency_repo(repository_name):
+        if repository_name == "origin" and "openshift-ansible" in repo_names:
+            return "openshift-ansible"
+        elif repository_name == "openshift-ansible" and "origin" in repo_names:
+            return "origin"
+        else:
+            return repository_name
+
     for repository in job_config.get("sync_repos", []):
         if repository.get("type", None) == "pull_request":
             sync_actions.append(PullRequestSyncAction(repository["name"]))
         else:
-            sync_actions.append(SyncAction(repository["name"]))
+            # if test is a PR, point to dependency repository to the PR's repository branch
+            repository_target_branch = get_dependency_repo(repository["name"]) if is_pull_request else repository["name"]
```

Comment:
I'm resolving it here... we dont know the branch name until the test is triggered. Upon the generation process we dont know whats the branch name, we have only info about what repos are synced and upon that we can assume what the `_TARGET_BRANCH` should be.
I've removed the additional action and to keep it simple since the `get_dependency_repo` will determine the right `repository_target_branch` ... also not really sure  that it makes sense to check  for the branch name in the `_SYNC_ACTION_TEMPLATE` since there should be always a matching version between `origin` and `o-a`